### PR TITLE
Add a new kusto version with changes

### PR DESCRIPTION
**Pull request by lucygoldbergmicrosoft**
The main changes in this PR:

1. Add a new version
2. Add two new resources, PrincipalAssignment for cluster and database
3. Add a new action diagnoseVirtualNetwork
4. Add a new property isFollowed for ReadWriteDatabaseProperties

### CODE REVIEW

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft is it a working PR? you can add label DO NOT MERGE instead of write in comment.

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , can you help me to figure out how to fix the kusto.json to make it pass the checks. It complains about something that was already published in the past and not related to my recent changes.

First it complains that the discriminator property Kind in Database definitions must be a required property. But After I added it as a required property, then it complains "Properties of a PATCH request body must not be required. PATCH operation: 'Databases_Update' Model Definition: 'Database' Property: 'kind'". 

Any idea howI can fix it?

**Comment by ChenTanyi** [comment]
> @ChenTanyi , can you help me to figure out how to fix the kusto.json to make it pass the checks. It complains about something that was already published in the past and not related to my recent changes.
> 
> First it complains that the discriminator property Kind in Database definitions must be a required property. But After I added it as a required property, then it complains "Properties of a PATCH request body must not be required. PATCH operation: 'Databases_Update' Model Definition: 'Database' Property: 'kind'".
> 
> Any idea howI can fix it?

@qiaozha could you take a look at the opposite requirement in CI.
@lucygoldbergmicrosoft is it possible for you to change any of them? Like change PATCH to PUT or do not use discriminator in swagger?

**Comment by qiaozha** [comment]
> @ChenTanyi , can you help me to figure out how to fix the kusto.json to make it pass the checks. It complains about something that was already published in the past and not related to my recent changes.
> First it complains that the discriminator property Kind in Database definitions must be a required property. But After I added it as a required property, then it complains "Properties of a PATCH request body must not be required. PATCH operation: 'Databases_Update' Model Definition: 'Database' Property: 'kind'".
> Any idea howI can fix it?
> 
> @qiaozha could you take a look at the opposite requirement in CI.
> @lucygoldbergmicrosoft is it possible for you to change any of them? Like change PATCH to PUT or do not use discriminator in swagger?

@ChenTanyi It's two different validation errors, when you don't add required kind, it breaks the semantic of swagger discriminator, when you add required kind, it breaks the "patch" operation requirement.
I think a better way to solve this problem is to create another definition without required property. Such as "DatabaseUpdate", you may refer to "patch" operation at line 115 of kusto.json. it uses "ClusterUpdate". And the "Cluster" has a required property.

**Comment by lucygoldbergmicrosoft** [comment]
@qiaozha, in a way, you are right. But DatabaseUpdate won't help me here because DatabaseUpdate still needs the discriminator kind (again, back to the same problem). ClusterUpdate did the trick because sku is not the discriminator. So my current solution is to suppress the error like it was done for DatabaseConnection

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , @qiaozha , please review it since all the checks passed. Thanks

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft you could delete the "DO NOT MERGE" in title

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , when can this be reviewed? It would also be great if you can help me to add back the DoNotMerge label. Thanks

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft new api needs ARM review first, I don't know when will they review. Do you need more change for the api now? If so, I can add DoNotMerge.

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , No more change For the API now. Thanks

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft so is there anything needed to add DoNotMerge?

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , sorry, misunderstood what you wrote. Our api is not 100% deployed yet. So we need the DoNotMerge flag to control when we publish it. Thanks

**Comment by KrisBash** [comment]
can you please schedule a meeting with armapireview to discuss principalAssignments?

**Comment by lucygoldbergmicrosoft** [comment]
hi, @KrisBash, thank you very much for responding. @radennis has done the armapireview ("Kusto Data plane Principals RP API review") for principalassignment design at August. And this api follows exactly what ARM suggested. So is there still a need to schedule a meeting to discuss it again? Thanks

**Comment by lucygoldbergmicrosoft** [comment]
@KrisBash, waiting for your response to confirm that there is no need for a review again. Thanks

**Comment by orhasban** [comment]
Yesterday we (Kusto team) did another review with @anthony-c-martin. This was the second review after the last one with @ryansbenson. 
We need your approval ASAP, and not sure why we are still blocked? 
The freeze is coming soon and we cannot wait till new year.

**Review by anthony-c-martin — COMMENTED** [review]
> Yesterday we (Kusto team) did another review with @anthony-c-martin. This was the second review after the last one with @ryansbenson.
> We need your approval ASAP, and not sure why we are still blocked?
> The freeze is coming soon and we cannot wait till new year.

Please note that we did not discuss this PR in any detail during yesterday's meeting - this was a design discussion for an independent feature.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: I see you've declared `x-ms-long-running-operation` - can this API return 202 then, or will it always complete synchronously?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Can this return 202 for async response?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Why do you need a location associated with this resource?

Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Same here - is location necessary?

Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Are you planning to add new types to this API in future?

Comment (anthony-c-martin) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Are you planning to add new types to this API in future?

**Review by anthony-c-martin — COMMENTED** [review]
Just a few questions for clarification.

File: custom-words.txt
Comment (lucygoldbergmicrosoft) [reply custom-words.txt]: this PR's automatic checks complain this word assignments. So I added the word here to solve the word error although I'm not sure why the checks complain

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: nope

Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: nope

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: This is an asynchronous API, but put never returns 202. Please check "Clusters_CreateOrUpdate"

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: put doesn't return 202 for async response. Please check Clusters_CreateOrUpdate

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json
Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Technically, no. But the RP SDK validates the location property for resources. If no location is provided, ResourceProviderException will be thrown

Comment (lucygoldbergmicrosoft) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-11-09/kusto.json]: Same

File: custom-words.txt
Comment (anthony-c-martin) [reply custom-words.txt]: It looks like a typo - "assginments" rather than "assignments". Has this been introduced in this PR? If so, please find and correct the misspelling.

Comment (lucygoldbergmicrosoft) [reply custom-words.txt]: You solved a mystery for me since I couldn't understand why it complains the word assignments. I really didn't notice the typo.

**Review by anthony-c-martin — APPROVED** [review]

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , please review this. Thanks

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft is it ready to merge?

**Comment by ChenTanyi** [comment]
@NullMDR could you take a look at sdk automation? Does it need to rerun?

File: specification/azure-kusto/resource-manager/readme.cli.md
Comment (ChenTanyi) [inline specification/azure-kusto/resource-manager/readme.cli.md]: Shoud change `\t` to `space`, align with the above.

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi, no, please don't merge. In the ARM preview, a few changes were required to change, which involve code changes. So we could not merge yet. Thanks

**Review by ChenTanyi — APPROVED** [review]

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , the relevant API changes are deployed to all clouds. Can you please help to merge this and publish the SDKs.

**Comment by lucygoldbergmicrosoft** [comment]
@anthony-c-martin, can you merge it? thanks a lot

**Comment by ChenTanyi** [comment]
@lucygoldbergmicrosoft I will merge it, and if you want to publish SDKs, you could send a SDK request in OpenAPI Hub. Thanks!

**Comment by lucygoldbergmicrosoft** [comment]
@ChenTanyi , thanks. Yes, I will publish the SDK.  Happy New Year!!!

**Pull request merged**
